app/repos/user-accounts.js

```diff
@@ -1,5 +1,111 @@
 const sequelize = require('../config/db')
 const { addMinutes } = require('../lib/date-helpers')
+const { DuplicateResourceError } = require('../errors/duplicate-record')
+const { getPoliceForce } = require('../lookups')
+const { NotFoundError } = require('../errors/not-found')
+const { createUserAccountAudit, deleteUserAccountAudit } = require('../dto/auditing/user')
+
+/**
+ * @typedef UserAccount
+ * @property {number} id
+ * @property {string} username
+ * @property {number|null} police_force_id - 1,
+ * @property {string|null} activation_token
+ * @property {Date|null} activation_token_expiry
+ * @property {Date|null} activated_date
+ * @property {Date|null} accepted_terms_and_conds_date
+ * @property {boolean} active
+ * @property {Date|null} last_login_date
+ * @property {Date} created_at
+ * @property {Date} updated_at
+ */
+
+/**
+ * @typedef UserAccountRequestDto
+ * @property {string} username
+ * @property {string} [telephone]
+ * @property {boolean} [active]
+ * @property {number} [police_force_id]
+ * @property {string} police_force
+ */
+
+/**
+ * @typedef UserAccountDto
+ * @property {string} username
+ * @property {string} [telephone]
+ * @property {boolean} [active]
+ * @property {number} [police_force_id]
+ */
+
+/**
+ * @param {UserAccountRequestDto} account
+ * @param user
+ * @param [transaction]
+ * @return {Promise<UserAccount>}
+ */
+const createAccount = async (account, user, transaction) => {
+  if (!transaction) {
+    return sequelize.transaction(async (t) => createAccount(account, user, t))
+  }
+
+  const foundUser = await sequelize.models.user_account.findOne({
+    where: {
+      username: account.username
+    }
+  }, transaction)
+
+  if (foundUser) {
+    throw new DuplicateResourceError('This user is already in the allow list')
+  }
+
+  const { police_force: policeForce, ...accountWithoutPoliceForce } = account
+
+  let createdAccount
+
+  if (!policeForce || account.police_force_id) {
```

Comment:
Should this be:
if (!policeForce || !account.police_force_id) {

Reply:
Not on this one...the logic is...

If there is no police force name declared (undefined), then we can call the create with the properties of account.
If police force id exists, then we can also call the create with the properties of account.

If both of these return false (ie - if there is no police_force_id but there is a police force name), we need to get the police force id from the DB.